### PR TITLE
[BTAT-8943] Predicate logic update for blocked and allowed insolvency…

### PR DESCRIPTION
**Pull request by daparkes**
… types

### CODE REVIEW

File: test/models/customerInformation/CustomerInformationSpec.scala [hunk 1]
```diff
@@ -141,4 +141,37 @@ class CustomerInformationSpec extends UnitSpec {
...
+
+  "Calling .isInsolventWithoutAccess" when {
+
+    "the user is insolvent and not continuing to trade" should {
+
+      "return true if there is no insolvency type" in {
+        customerInfoInsolvent.isInsolventWithoutAccess shouldBe true
+      }
+
+      "return false if the insolvency type is allowed" in {
```
Comment (paulfirmstone) [inline test/models/customerInformation/CustomerInformationSpec.scala:153]: Can we use a foreach loop to test all of the insolvency types?

File: test/models/customerInformation/CustomerInformationSpec.scala [hunk 2]
```diff
@@ -141,4 +141,37 @@ class CustomerInformationSpec extends UnitSpec {
...
+        customerInfoInsolvent.copy(insolvencyType = Some("07")).isInsolventWithoutAccess shouldBe false
+      }
+
+      "return true if the insolvency type is blocked" in {
+        customerInfoInsolvent.copy(insolvencyType = Some("08")).isInsolventWithoutAccess shouldBe true
```
Comment (paulfirmstone) [inline test/models/customerInformation/CustomerInformationSpec.scala:158]: Can we use a foreach loop to test all of the insolvency types?

File: test/models/customerInformation/CustomerInformationSpec.scala [hunk 3]
```diff
@@ -141,4 +141,37 @@ class CustomerInformationSpec extends UnitSpec {
...
+    }
+
+    "the user is insolvent and is continuing to trade" should {
+
+      "return false for a user with no insolvency type" in {
+        customerInfoInsolventContinueToTrade.isInsolventWithoutAccess shouldBe false
+      }
+
+      "return false for a user with an allowed insolvency type" in {
+        customerInfoInsolventContinueToTrade.copy(insolvencyType = Some("12")).isInsolventWithoutAccess shouldBe false
```
Comment (paulfirmstone) [inline test/models/customerInformation/CustomerInformationSpec.scala:169]: Can we use a foreach loop to test all of the insolvency types?

File: test/models/customerInformation/CustomerInformationSpec.scala [hunk 4]
```diff
@@ -141,4 +141,37 @@ class CustomerInformationSpec extends UnitSpec {
...
+      }
+
+      "return true for a user with a blocked insolvency type" in {
+        customerInfoInsolventContinueToTrade.copy(insolvencyType = Some("09")).isInsolventWithoutAccess shouldBe true
```
Comment (paulfirmstone) [inline test/models/customerInformation/CustomerInformationSpec.scala:173]: Can we use a foreach loop to test all of the insolvency types?

File: test/models/customerInformation/CustomerInformationSpec.scala [hunk 5]
```diff
@@ -141,4 +141,37 @@ class CustomerInformationSpec extends UnitSpec {
...
+      }
```
Comment (paulfirmstone) [inline test/models/customerInformation/CustomerInformationSpec.scala:174]: Can you add tests for when a continueToTrade flag is not returned?

**Review by paulfirmstone — CHANGES_REQUESTED** [review]

File: app/models/customerInformation/CustomerInformation.scala [hunk 6]
```diff
@@ -58,9 +59,19 @@ case class CustomerInformation(ppob: PPOB,
...
-    case _ => false
+  val allowedInsolvencyTypes: Seq[String] = Seq("07", "12", "13", "14")
+  val blockedInsolvencyTypes: Seq[String] = Seq("08", "09", "10", "15")
+
+  val isInsolventWithoutAccess: Boolean = {
+    if(isInsolvent) {
+      insolvencyType match {
+        case Some(insolvencyType) if allowedInsolvencyTypes.contains(insolvencyType) => false
+        case Some(insolvencyType) if blockedInsolvencyTypes.contains(insolvencyType) => true
+        case _ => !continueToTrade.getOrElse(true)
```
[the remark this answers is not in the document]
Comment (daparkes) [reply app/models/customerInformation/CustomerInformation.scala:70]: Yes!

**Pull request merged**